hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -184,6 +192,35 @@ public static void verifyResourceName(String resName) {
     }
   }
 
+  /**
+   * verifies that bucket name / volume name is a valid DNS name.
+   *
+   * @param resName Bucket or volume Name to be validated
+   *
+   * @throws IllegalArgumentException
+   */
+  public static void verifyResourceName(String resName, boolean isNamespaceS3RuleCompliant) {
+
+    doNameChecks(resName);
+
+    boolean isIPv4 = true;
+    char prev = (char) 0;
+
+    for (int index = 0; index < resName.length(); index++) {
+      char currChar = resName.charAt(index);
+      if (currChar != '.') {
+        isIPv4 = ((currChar >= '0') && (currChar <= '9')) && isIPv4;
+      }
+      doCharacterChecks(currChar, prev, isNamespaceS3RuleCompliant);
+      prev = currChar;
+    }
+
+    if (isIPv4) {
+      throw new IllegalArgumentException(
+              "Bucket or Volume name cannot be an IPv4 address or all numeric");
+    }
+  }
```

Comment:
The only difference is in the call to `doCharacterChecks`, where one function passes `true`, the other passes the argument `isNamespaceS3RuleCompliant`.  Instead of duplicating the entire function, `verifyResourceName(String)` can simply call `verifyResourceName(String, boolean)` with `true`.

Reply:
Yup good catch! updated.